### ISSUE TITLE
Prisma Client errors with "TypeError: Cannot create proxy with a non-object as target or handler" when using result client extension with no `needs` and `count` method

### ISSUE THREAD

**Issue by semoal** [prisma/prisma#20499]
### Bug description

Just upgraded to 5.1.0, and after adjusting the Array shortcut breaking change, I have run the testing suite and this test fails, I'm not quite sure how to debug this.

Any help or question is appreciated and I would love to help.

```bash
 FAIL  src/controllers/v1/users.controller.test.ts > Users > Search > should search
TypeError: Cannot create proxy with a non-object as target or handler
 ❯ xe ../../packages/prisma-client/client/runtime/library.js:26:5353
 ❯ Ds ../../packages/prisma-client/client/runtime/library.js:112:2696
 ❯ visitor ../../packages/prisma-client/client/runtime/library.js:112:3823
 ❯ $r ../../packages/prisma-client/client/runtime/library.js:112:3062
 ❯ Is ../../packages/prisma-client/client/runtime/library.js:112:3751
 ❯ l ../../packages/prisma-client/client/runtime/library.js:126:10330
 ❯ search src/controllers/v1/users.controller.ts:149:17
    147|     : { name: Prisma.SortOrder.asc };
    148|
    149|   const count = await prisma.user.count();
```

Our User schema is quite big so it's complicated to share.

I'm using Postgres, Node 18.13.0, Typescript 5.0.4 and a PNPM monorepo if it helps. 

### How to reproduce

-

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: Macosx
- Database: Postgresql
- Node.js version: 18.13.0

### Prisma Version

```
5.1.0
```

**Comment by semoal** [comment]
Yes, exactly

**Comment by semoal** [comment]
The previous version I was of Prisma its: `4.15.0`, and Vitest I'm using 0.34.1 (latest) but tried with 0.32.0 and got the same error

**Comment by janpio** [comment]
The error message also mentions `: { name: Prisma.SortOrder.asc };`, while the code you shared does not. Might that be relevant?
Can you maybe isolate the schema and code into its own, standalone project so you can see if you can reproduce it? Sharing that project with us would be perfect for us to reproduce, understand and hopefully fix.

**Comment by semoal** [comment]
> The error message also mentions `: { name: Prisma.SortOrder.asc };`, while the code you shared does not. Might that be relevant?
> 
> Can you maybe isolate the schema and code into its own, standalone project so you can see if you can reproduce it? Sharing that project with us would be perfect for us to reproduce, understand and hopefully fix.

I'll try to isolate but it's a large repo so it's a bit complex, but will try for sure.
About the Prisma.SortOrder.asc, was not relevant tried with the code I've pasted here and got the same error.

**Comment by semoal** [comment]
Got it guys!

**Comment by SevInf** [comment]
Reopening, since it is an actual bug in 5.1. Even though there is a workaround for it (thank you for providing it, @semoal ), we need to fix it.

**Comment by SevInf** [comment]
Fix released in 5.1.1
